src/builder.rs

```diff
@@ -27,6 +27,46 @@ use crate::{
 
 const MIN_ACTIONS: usize = 2;
 
+/// An enumeration of rules for construction of dummy actions that may be applied to Orchard bundle
+/// construction.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum PaddingRule {
+    /// A rule that requires that at least the specified number of Orchard actions be constructed,
+    /// irrespective of whether any genuine actions are included.
+    Require(usize),
+    /// A rule that requires that at least the specified number of Orchard actions be constructed,
+    /// iff any genuine actions are included.
+    PadTo(usize),
+    /// A padding rule that specifies that no additional dummy Orchard actions are to be
+    /// constructed.
+    None,
+}
+
+impl PaddingRule {
+    /// The default padding rule, which ensures that the constructed bundle will contain at least 2
+    /// actions if any genuine Orchard spends or outputs are requested.
+    pub const DEFAULT: PaddingRule = PaddingRule::PadTo(MIN_ACTIONS);
+
+    /// Returns the number of logical actions that the builder will add to the bundle
+    /// as a consequence of this padding rule, given the specified number of spends
+    /// and outputs.
+    pub fn num_actions(&self, num_spends: usize, num_outputs: usize) -> usize {
+        let num_real_actions = core::cmp::max(num_spends, num_outputs);
+
+        match self {
+            PaddingRule::Require(n) => core::cmp::max(num_real_actions, *n),
+            PaddingRule::PadTo(n) => {
+                if num_real_actions == 0 {
+                    0
```

Comment:
This introduces the possibility of a panic: if the builder uses this padding rule and is given no spends and no outputs, then `Builder::build` will panic when `NonEmpty` is constructed.

If we want to support this, then we need to change `Builder::build` to return `Option<Bundle>` (which would bring it closer to the `sapling-crypto` API.

Reply:
Done in 99d3d0ab71c3f6c8afbccb3b8f10213fe9dc2199